### DISCUSSION TITLE
Current Setup - Possible suggestions?

### DISCUSSION THREAD

**Discussion by rpkilgore** [roleoroleo/yi-hack-Allwinner-v2#373] [Ideas]
This might be the wrong forum to ask this but I'm looking for some possible alternatives to my current setup.  I'd like to sortof duplicate what the kami app does but not sure if that's possible.  Would like some way to be notified of motion, then be sent either a 10sec clip or the whole vid to review (Maybe HomeAssistant could do this with the MQTT or ONVIF thing?  I have 0 experience with these protocols/servies)  Also, I was curious what others on here have done with their Kami/Yi Cams. 

**Current Setup**
Currently, My cams all upload using ftp to my [TrueNas ](https://www.truenas.com/) server, then I have a jail setup that runs an automation script I wrote to filter out videos without motion using [DVRScan](https://dvr-scan.readthedocs.io/en/latest/) , this outputs an avi file with a name of 'cam+folder+filename' so if I want to see more  I can go back to the cam and download the full video plus before and after vids if I need them.  Then finally I have a CloudSync task setup to upload the avi files to my onedrive account that way I can view the avi's from my phone or computer.

This works pretty well because I've noticed the ftp function seems to remember what it has uploaded and doesn't duplicate very often.  The reason I have to filter with [DVRScan ](https://dvr-scan.readthedocs.io/en/latest/) is because the Motion Events seems unreliable.  Sometimes it will record a 1 min video every min for an hour with no motion, other times it will only record 1 or 2 videos but they have motion.   DVRScan works fairly well to trim down to videos that actually have real motion, but it doesn't have any real like AI human detection so sometimes tress blowing or shadows moving are captured.

Thanks ahead of time for any idea's and for sharing your current setups. :)

**Comment by Giuserver** [comment]
You have all tools for replicate kami  app with this hack.
Mqtt (for trigger) 
Onvif
FtpClient/Server
ssh
etc....
but here you cannot find a similar app. 
you have to programming it!

**Reply by rpkilgore** [reply]
lol cool thanks.. My programming days are long over, but I might dust them off to come up with something.. My current setup seems to work ok, but I think I could make better use of mqtt or onvif..

**Reply by Giuserver** [reply]
> lol cool thanks.. My programming days are long over, but I might dust them off to come up with something.. My current setup seems to work ok, but I think I could make better use of mqtt or onvif..

yes you can. I'll download your app!

**Comment by uklatt** [comment]
You can use an external software like Frigate to save and analyze the RTSP video stream.
Frigate uses MQTT to send alarm messages...

Uwe

**Reply by rpkilgore** [reply]
Thanks! I'll check that out..
